pkg/scheduler/internal/cache/cache.go

```diff
@@ -280,9 +280,11 @@ func (cache *schedulerCache) updateNodeInfoSnapshotList(snapshot *Snapshot, upda
 	if updateAll {
 		// Take a snapshot of the nodes order in the tree
 		snapshot.nodeInfoList = make([]*framework.NodeInfo, 0, cache.nodeTree.numNodes)
-		cache.nodeTree.resetExhausted()
-		for i := 0; i < cache.nodeTree.numNodes; i++ {
-			nodeName := cache.nodeTree.next()
+		nodesList, err := cache.nodeTree.list()
+		if err != nil {
+			klog.Error(err.Error())
```

Comment:
change this to `Fatal` to align with what you did in the `cache_test.go`?

Reply:
because it is not actually fatal for the scheduler run. It is fatal in the tests as it should not happen at all, but for the logging, from the user perspective, it should not be fatal for the scheduler, and not perceived as such. It just means we got less nodes than we expected.

Reply:
Go you, thank!